### ISSUE TITLE
Importing hourly data

### ISSUE THREAD

**Issue by jbrider** [APSIMInitiative/ApsimX#5930]
@HamishBrownPFR @hol353 @hut104  @BrianCollinss @hol430 

We are wanting to import hourly temperature and radiation data from a file for use by DCaPS.
The ability to interpolate hourly data was added as per #5121 
I didn't miss an implementation that read it from a file as well did I?

**Comment by hut104** [comment]
If we do something here, it should be carefully planned.  For example, it should perhaps cover all weather data (not just temperature and radiation) and data on any consistent time step.

________________________________

**Comment by BrianCollinss** [comment]
Jason, as fas as I remember Hamish was implementing some algorithms and he took one from our work. I'm not sure if it was temp or radiation. None could read from files.

**Comment by HamishBrownPFR** [comment]
Hi Guys.  I implemented hourly temperature and radiation interpolation methods in Models/Functions/HourlyInterpolation.cs.  Neither of these read from file and they just interpolate to hourly values then return a daily sum or average so they are working at the same time step as the rest of the model externally.
There are bound to be loads of things that would use hourly met data if it were read in from a file and available to other models that work at an hourly time step.  Weirdo and Swim for example could use hourly rainfall and ET data if it were there.  So it would be good to design things carefully.  
Would it be good to divorce the supply of hourly met data from the demand.  So the met object is capable of providing daily or hourly data depending on what individual models demand.  Where hourly data is provided from a met file that is used and where it is not provided some interpolation is applied.  That way models needing hourly data will still run if only daily data is available in the met file but where specific hourly data is desired this can be provided by the met file.

**Comment by jbrider** [comment]
@HamishBrownPFR That describes what we have started in Classic pretty well. We created a met object that when asked for hourly data it either provides the data from a file, or calculates it using interpolation. In Nextgen it should be easier to allow dropping an interpolation function on the Met component to determine how the interpolation works. It could be a different method for temperature, compared to radiation - and others as @hut104  suggested.
